addons/base_import/models/base_import.py

```diff
@@ -1459,9 +1464,39 @@ def execute_import(self, fields, columns, options, dryrun=False):
         # convert load's internal nextrow to the imported file's
         if import_result['nextrow']: # don't update if nextrow = 0 (= no nextrow)
             import_result['nextrow'] += skip
+        if additional_filenames:
+            import_result['additional_filenames'] = additional_filenames
 
         return import_result
 
+    def _extract_additional_filenames(self, import_fields, data, model=False, prefix='', additional_filenames=False):
+        model = model or self.res_model
+        additional_filenames = additional_filenames or {}
+        all_fields = self.env[model].fields_get()
+        for name, field in all_fields.items():
+            name = prefix + name
+            if any(name + '/' in import_field and name == import_field.split('/')[prefix.count('/')] for import_field in import_fields):
+                # Recursive call with the relational as new model and add the field name to the prefix
+                data, additional_filenames = self._extract_additional_filenames(import_fields, data, field['relation'], name + '/', additional_filenames)
+            elif field['type'] == 'binary' and field.get('attachment') and any(f in name for f in IMAGE_FIELDS) and name in import_fields:
+                additional_filenames[name] = []
+                index = import_fields.index(name)
+                for line in data:
+                    filename = None
+                    value = line[index]
+                    if isinstance(value, str):
+                        if re.match(config.get("import_image_regex", DEFAULT_IMAGE_REGEX), value):
+                            pass
+                        elif value.find('.') >= 0:
+                            # Detect if it's a filename
+                            filename = value
+                            line[index] = ''
+                        else:
+                            # base64
+                            pass
+                    additional_filenames[name].append(filename)
```

Comment:
Simplification suggestion (I changed the fact that you sometimes add None to `additional_filenames[name]` for no reason:
```suggestion
                for line in data:
                    value = line[index]
                    if not isinstance(value, str):
                        continue
                    if re.match(config.get("import_image_regex", DEFAULT_IMAGE_REGEX), value):
                        continue
                    if '.' in value:
                        # Detect that's a filename
                        additional_filenames[name].append(value)
                        line[index] = ''
```

I didn't test it :grimacing:

Reply:
I have a reason to add `None` when there is nothing, it's to link the file name to the row of the import as when the import is ended the only informations I had in JavaScript is an array of `id` of the created records.

So in JavaScript i can loop the result and pair it with the id.

e.g.:
```javascript
const { ids, binary_filenames } = await this.orm.call("base_import.import", "execute_import", ...);
const fields = Object.keys(binary_filenames);
for (let rowIndex = 0; rowIndex < ids.length; rowIndex++) {
    const id = ids[rowIndex];
    for (const field of fields) {
        const fileName = binary_filenames[field][rowIndex];
```